h5netcdf/core.py

```diff
@@ -157,9 +230,28 @@ def __getitem__(self, key):
             string_info = h5py.check_string_dtype(self._h5ds.dtype)
             if string_info and string_info.length is None:
                 return self._h5ds.asstr()[key]
+
+        # return array padded with fillvalue
+        if self.dtype != str and self.dtype.kind in ["f", "i", "u"]:
```

Comment:
To things are currently unclear to me:

- is `["f", "i", "u"]` enough for now?
- is `np.pad` the right function for this?